packages/core/docs/recipes/validation.md

```diff
@@ -0,0 +1,286 @@
+  ms-validation adding schema [todo.list], /create-microfleet-app/schemas/todo.list.json with id choice of $id: [todo.list] vs defaultName: [todo.list] +0ms
+
+```
+
+
+### Request validation
+Our service will respond to POST `http://localhost/todo/create` URL and validate Request using  [`todo.create`](#create-todo) schema. The easiest way is to define your actions `schema` property, and `Microfleet` will do the rest of the work.
+
+Let's create a new `src/actions/todo/create.js` file with contents:
+```js
+const { ActionTransport } = require('@microfleet/core');
+
+function createTodo(request) {
+  const { todo } = request.params;
+  // Create our new TODO....
+  return { ok: { todo }};
+}
+
+createTodo.transports = [ActionTransport.http];
+// Schema used for request validation
+createTodo.schema = 'todo.create';
+module.exports = createTodo;
+
+```
+
+Now you may start your microservice and check that validation is working:
+
+```console
+$@> curl http://0.0.0.0:3000/todo/create -H "Content-Type: application/json" -X POST --data '{"todo":{"name" : "foo"}}' | jq
+```
+And the response is goint to be like:
+```json
+{
+  "statusCode": 400,
+  "error": "Bad Request",
+  "message": "todo.create validation failed: data.todo should have required property 'state'",
+  "name": "HttpStatusError"
+}
+```
+
+### Query string validation
+In some cases, you will need to validate incoming Query string parameters.
+`Microfleet` parsing your QS parameters, BUT not doing type conversion, so the default is `string`.
+
+If you want to perform typed validation, you must enable an additional Router extension called `query-string-parser`.
+
+#### 1. Enable `query-string-parser`
+Update your configuration for `Router`:
+  * Add `query-string-parser` plugin.
+  * Enable `preValidate` event.
+```js
+const { routerExtension } = require('@microfleet/core');
+const qsValidator = routerExtension('validate/query-string-parser');
+
+exports.router = {
+  extensions: {
+    enabled: [ 'preValidate' ],
+    register: [ qsValidator ],
+  },
+};
+```
+
+#### 2. Create `/todo/list` action
+Our Service will respond to `GET https://localhost:3000/todo/list?page=10&hidden=1`.
+Assuming that `page` should have type `integer` and `hidden` will accept `0|1|true|false` converted to `boolean`. 
+
+Let's create new file `src/actions/todo/list` with contents:
+```js
+const { ActionTransport } = require('@microfleet/core');
+
+function listTodo(request) {
+  // Well here should be `list` code, but we will respond with object containing our query parameters
+  return { query : request.query};
+}
+
+listTodo.schema = 'todo.list';
+listTodo.transports = [ActionTransport.http];
+// Our action responds only on GET method.
+listTodo.transportsOptions = {
+  [ActionTransport.http]: {
+    methods: ['get'],
+  },
+};
+
+// Function used by `query-string-parser` to transform incoming QS parameters.
+listTodo.transformQuery = (query) => {
```

Comment:
Agreed on stashing it to use in a recipe dedicated to the query string parser extension